drivers/counter/counter_native_posix.c

```diff
@@ -84,15 +99,40 @@ static int ctr_set_top_value(const struct device *dev,
 			     const struct counter_top_cfg *cfg)
 {
 	ARG_UNUSED(dev);
-	ARG_UNUSED(cfg);
 
-	posix_print_warning("%s not supported\n", __func__);
-	return -ENOTSUP;
+	if (is_alarm_pending) {
+		posix_print_warning("Can't set top value while alarm is active\n");
+		return -EBUSY;
+	}
+
+	if (cfg->flags & COUNTER_TOP_CFG_DONT_RESET) {
+		if (hw_counter_get_value() > cfg->ticks) {
+			if (cfg->flags & COUNTER_TOP_CFG_RESET_WHEN_LATE) {
+				hw_counter_reset();
+			}
+			return -ETIME;
+		}
+	} else {
+		hw_counter_reset();
+	}
+
+	top = *cfg;
+
+	if ((cfg->ticks != TOP_VALUE) && cfg->callback) {
+		is_top_set = true;
+		hw_counter_set_target(cfg->ticks);
+		irq_enable(COUNTER_EVENT_IRQ);
+	} else {
+		is_top_set = false;
+		irq_disable(COUNTER_EVENT_IRQ);
```

Comment:
If you disable the interrupt, you would not handle the 32bit wrap around around TOP_VALUE
I guess you should also set a new target for the counter in this case to TOP_VALUE, or?

Reply:
see other thread